osbs/api.py

```diff
@@ -314,25 +362,30 @@ def _create_build_config_and_build(self, build_request):
             # Reset name change that may have occurred during
             # update above, since renaming is not supported.
             existing_bc['metadata']['name'] = build_config_name
-            logger.debug('build config for %s already exists, updating...', build_config_name)
+            logger.debug('build config for %s already exists, updating...',
+                         build_config_name)
             self.os.update_build_config(build_config_name, json.dumps(existing_bc))
 
+        if image_stream:
+            changed_ist = self.ensure_image_stream_tag(image_stream,
+                                                       image_stream_tag_name,
+                                                       scheduled=True)
+            logger.debug('Changed parent ImageStreamTag? %s', changed_ist)
+
+        if not existing_bc:
```

Comment:
It's actually in this order for a reason. created.

If the BC already exists and an update is to be performed, we want to update/create IST after BC has been updated. Creating IST will trigger the build instantiation and we want to make sure it's using the latest and greatest BuildConfig.

If the BC does not exist, then IST needs to be created first. This is to avoid two builds being auto instantiated (one for BC creation and one for IST creation). Creating IST first wouldn't trigger build instantiation because well.. BuildConfig doesn't exist yet! But once BuildConfig is created, then it would be auto instantiated as expected.